### PR TITLE
cmake package config installation

### PR DESCRIPTION
**Pull request by headupinclouds**
find_package(BoringSSL CONFIG REQUIRED)
target_link_libraries(foo BoringSSL::ssl BoringSSL::crypto)

* BoringSSL::ssl
* BoringSSL::crypto
* BoringSSL::decrepit
* BoringSSL::Fuzzer

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,19 @@
 cmake_minimum_required (VERSION 2.8.11)
+cmake_policy(SET CMP0048 NEW)
```
Comment (ruslo) [inline CMakeLists.txt]: Policy introduced in 3.0: https://cmake.org/cmake/help/latest/policy/CMP0048.html
But minimum version set to 2.8. If user will try to run with CMake 2.8 he will hit error.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
 
 # Defer enabling C and CXX languages.
-project (BoringSSL NONE)
+project (BoringSSL VERSION 1.0.0 LANGUAGES NONE)
```
Comment (ruslo) [inline CMakeLists.txt]: `project(... VERSION ...)` available since CMake 3.0 => `cmake_minimum_required(VERSION 3.0)` have to be used.

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (headupinclouds) [reply CMakeLists.txt]: Done.

Comment (ruslo) [reply CMakeLists.txt]: ?

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (headupinclouds) [reply CMakeLists.txt]: Done.

Comment (headupinclouds) [reply CMakeLists.txt]: https://github.com/hunter-packages/boringssl/pull/1/commits/301c51321506c1f3e9c4ebbbfcc53f8d71227f02

**Review by ruslo — APPROVED** [review]

**Review by ruslo — APPROVED** [review]
Looks good

**Comment by headupinclouds** [comment]
If this looks good we can merge it as an initial CMake + install patch.  A follow up PR to make tests optional will save a fair amount of build time in the context of the hunter (lib only) use case.

**Comment by headupinclouds** [comment]
Thanks for the review.

**Comment by headupinclouds** [comment]
Note: I don't think boringssl uses [semantic versioning](http://semver.org) anywhere.  I've searched the source files, build files and github tags/releases without finding anything appropriate.  I've tagged this one as 1.0.0 to mark the initial hunter-package fork release.

**Pull request merged**